graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/GlobalBuiltins.java

```diff
@@ -1284,6 +1300,50 @@ public boolean isCallerSensitive() {
         }
     }
 
+    /**
+     * Implementation of Module Block serialize-method
+     */
+    public abstract static class JSGlobalSerializeNode extends JSBuiltinNode {
+
+        protected JSGlobalSerializeNode(JSContext context, JSBuiltin builtin) {
+            super(context, builtin);
+        }
+
+        protected byte[] serialize(Object value) {
+            if (DynamicObjectLibrary.getUncached().containsKey((DynamicObject) value, (Object) ModuleBlockNode.getModuleBodyKey())) {
+                PropertyGetNode getSourceCode = PropertyGetNode.createGetHidden(ModuleBlockNode.getModuleSourceKey(), getContext());
+
+                Object sourceCode = getSourceCode.getValue(value);
+
+                return ("module {" + sourceCode.toString() + "}").getBytes();
+            }
+
+            // TODO Errors.createTypeError("Not a ModuleBlock");
```

Comment:
`throw Errors.createTypeError("Not a ModuleBlock");` should work here :-)

Reply:
Reactivated.